test/suite/test_encrypt05.py

```diff
@@ -0,0 +1,80 @@
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_encrypt03.py
+#   Test some error conditions with quoted escaped characters.
+#
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# Test error conditions with quoted escaped characters.
+class test_encrypt05(wttest.WiredTigerTestCase):
+
+    types = [
+        ('table', dict(uri='table:test_encrypt05')),
+    ]
+    encrypt = [
+        ('rotn', dict( sys_encrypt='rotn', sys_encrypt_args=',keyid=11',
+            file_encrypt='rotn', file_encrypt_args=',keyid=13')),
+    ]
+    scenarios = make_scenarios(types, encrypt)
+
+    def conn_extensions(self, extlist):
+        extlist.skip_if_missing = True
+        extlist.extension('encryptors', self.sys_encrypt)
+        extlist.extension('encryptors', self.file_encrypt)
+
+    def conn_config(self):
+        return 'encryption=(name={0}{1}),'.format(
+            self.sys_encrypt, self.sys_encrypt_args)
+
+    # Open connection with a config that has a malformed keyid.
+    # The keyid is malformed because it contains a quoted escaped character.
+    def test_encrypt(self):
+        escaped_chars = ['\n', '\r', '\t', '\b']
+        for escaped_char in escaped_chars:
+            sys_encrypt = 'rotn'
+            sys_encrypt_args = f',keyid=\"11{escaped_char}\"' # Intentionally malformed keyid
+            config = 'encryption=(name={0}{1}),'.format(
+                sys_encrypt, sys_encrypt_args)
+
+            try:
+                self.reopen_conn(config=config)
+            except wiredtiger.WiredTigerError as e:
+                # If we get an error, it should be about the malformed keyid.
+                self.assertTrue('Invalid argument' in str(e), f"Unexpected error: {e}")
+
+            continue
```

Comment:
No need to have continue. Because you caught the expection already.

Reply:
The continue here is to let each of the escaped characters (4 in total) go through `reopen_conn` and caught exception, so that we verify all those escaped characters.

Reply:
Actually you are right the continue is not necessary here, will remove it. In an old version of the test there was logics added after the continue, which was removed afterwards - my apology that my mind still stayed around that time :)

Reply:
See [44e64d9](https://github.com/wiredtiger/wiredtiger/pull/12194/commits/44e64d9cda4a1e58a2c348838efa0e31a061e196)